File: android-sdk/sdk/src/main/java/com/adlcom/mxl/sdk/logging/Logger.kt
package com.adlcom.mxl.sdk.logging

import android.content.Context
import android.util.Log
import com.adlcom.mxl.sdk.config.SdkConfiguration

/**
 * SDK logger for internal logging.
 */
object Logger {
    private const val TAG = "MxLSdk"
    private var isInitialized = false
    private var enableDebug = false
    
    fun initialize(context: Context, configuration: SdkConfiguration) {
        isInitialized = true
        // Check if debug mode using reflection
        try {
            val buildConfigClass = Class.forName("com.adlcom.mxl.sdk.BuildConfig")
            val debugField = buildConfigClass.getField("DEBUG")
            enableDebug = debugField.getBoolean(null) as Boolean
        } catch (e: Exception) {
            enableDebug = false
        }
    }
    
    fun d(message: String) {
        if (enableDebug) {
            Log.d(TAG, message)
        }
    }
    
    fun i(message: String) {
        Log.i(TAG, message)
    }
    
    fun w(message: String) {
        Log.w(TAG, message)
    }
    
    fun e(message: String, throwable: Throwable? = null) {
        Log.e(TAG, message, throwable)
    }
}

File: ios-sdk/Sources/MxLSdk/Logging/Logger.swift
import Foundation
import Logging

/**
 * SDK logger for internal logging.
 */
internal class Logger {
    private let logger: Logging.Logger
    private static var sharedInstance: Logger?
    
    static var shared: Logger {
        return sharedInstance ?? Logger(logger: Logging.Logger(label: "com.adlcom.mxl.sdk"))
    }
    
    static func initialize(configuration: SdkConfiguration) -> Logger {
        var logger = Logging.Logger(label: "com.adlcom.mxl.sdk")
        logger.logLevel = .info
        let instance = Logger(logger: logger)
        sharedInstance = instance
        return instance
    }
    
    private init(logger: Logging.Logger) {
        self.logger = logger
    }
    
    func debug(_ message: String) {
        logger.debug("\(message)")
    }
    
    func info(_ message: String) {
        logger.info("\(message)")
    }
    
    func warning(_ message: String) {
        logger.warning("\(message)")
    }
    
    func error(_ message: String, error: Error? = nil) {
        if let error = error {
            logger.error("\(message): \(error.localizedDescription)")
        } else {
            logger.error("\(message)")
        }
    }
}

